src/devices/ExplorerHat/Lighting/Led.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Device.Gpio;
+using ExplorerHat.Gpio;
+
+namespace Iot.Device.ExplorerHat.Lighting
+{
+    /// <summary>
+    /// Represents a led light
+    /// </summary>
+    public class Led : IDisposable
+    {
+        private GpioController _controller;
+
+        /// <summary>
+        /// GPIO pin to which led is attached
+        /// </summary>
+        public int Pin { get; private set; }
+
+        /// <summary>
+        /// Led number on Hat
+        /// </summary>
+        public int Number { get; private set; }
+
+        /// <summary>
+        /// Led name on Hat
+        /// </summary>
+        public string Name { get; private set; }
+
+        /// <summary>
+        /// Gets if led is switched on or not
+        /// </summary>
+        /// <value></value>
+        public bool IsOn { get; set; }
+
+        /// <summary>
+        /// Initializes a <see cref="Led"/> instance
+        /// </summary>
+        /// <param name="number">Led number</param>
+        /// <param name="name">Led name</param>
+        /// <param name="pin">Underlying rpi GPIO pin number</param>
+        /// <param name="controller"><see cref="GpioController"/> used by <see cref="Led"/> to manage GPIO resources</param>
+        internal Led(int number, string name, int pin, GpioController controller)
+        {
+            _controller = controller;
+            Number = number;
+            Name = name;
+            Pin = pin;
+            IsOn = false;
+        }
+
+        /// <summary>
+        /// Switch on this led light
+        /// </summary>
+        public void On()
+        {
+            if (!IsOn)
+            {
+                _controller.EnsureOpenPin(Pin, PinMode.Output);
```

Comment:
can you open all pins in the constructor, we normally don't late open pins since most of the time people don't replug during runtime anyway

Reply:
also we never ensured pins are opened, we simply tried to open them, I'd follow this pattern in here as well